### PR TITLE
feat(ripple): Add support for activated and selected states

### LINKED ISSUE #1713: Update states to handle activated/selected state
The states subsystem should handle activated and selected states, in addition to the existing hover, focus, and pressed states

### PR DESCRIPTION
**Pull request by kfranqueiro**
Related to #1556; resolves #1713.

This adds support for optionally specifying a color for activated and/or selected states in the simple `mdc-states` mixin. If the values are not present, the styles are not emitted for these modifiers.

In cases where you're using the advanced mixins, you would simply re-apply those mixins within e.g. a `.foo--activated` or `.foo--selected` selector.

### CODE REVIEW

**Comment by lynnjepsen** [comment]
Resolves #1713

File: packages/mdc-ripple/_mixins.scss [hunk 1]
```diff
@@ -169,6 +168,41 @@ $mdc-ripple-common-styles-emitted_: false !default;
   @include mdc-states-hover-opacity(map-get($opacity-map, "hover"));
   @include mdc-states-focus-opacity(map-get($opacity-map, "focus"), $has-nested-focusable-element);
   @include mdc-states-press-opacity(map-get($opacity-map, "press"));
+
+  @if $activated-color {
```
Comment (lynnjepsen) [inline packages/mdc-ripple/_mixins.scss]: Can you create a `mdc-activated-state($color, $has-nested-focusable-element)` mixin which just handles activated state? It seems like its isolated logic

File: packages/mdc-ripple/_mixins.scss [hunk 2]
```diff
@@ -169,6 +168,41 @@ $mdc-ripple-common-styles-emitted_: false !default;
...
+    &--activated {
```
Comment (lynnjepsen) [inline packages/mdc-ripple/_mixins.scss]: does --activated work with all existing components? Or will this be a new modifier for some components?

File: packages/mdc-ripple/_mixins.scss [hunk 3]
```diff
@@ -169,6 +168,41 @@ $mdc-ripple-common-styles-emitted_: false !default;
...
+      @include mdc-states-hover-opacity(map-get($opacity-map, "activated") + map-get($opacity-map, "hover"));
+      @include mdc-states-focus-opacity(
+        map-get($opacity-map, "activated") + map-get($opacity-map, "focus"),
+        $has-nested-focusable-element
+      );
+      @include mdc-states-press-opacity(map-get($opacity-map, "activated") + map-get($opacity-map, "press"));
+    }
+  }
+
+  @if $selected-color {
```
Comment (lynnjepsen) [inline packages/mdc-ripple/_mixins.scss]: Can you create a mdc-selected-state($color, $has-nested-focusable-element) mixin which just handles selected state? It seems like its isolated logic

File: packages/mdc-ripple/_mixins.scss [hunk 4]
```diff
@@ -169,6 +168,41 @@ $mdc-ripple-common-styles-emitted_: false !default;
...
+    &--selected {
```
Comment (lynnjepsen) [inline packages/mdc-ripple/_mixins.scss]: does --selected work with all existing components? Or will this be a new modifier for some components?

**Comment by kfranqueiro** [comment]
I've updated the branch to separate the basic mixin into 3 (one for normal interaction states, one for when a surface is activated, and one for when it is selected).

RE the questions around working with all existing components, no, these will be used in the future by components intended to support these states. Currently drawer and menu have some modifiers for selected state, but those need to be tweaked a bit because they're modifying the wrong block. Andy has also done some work in https://github.com/material-components/material-components-web/commit/6ea948f266b340f6733e3b005c0e25e93028955c to make way for the activated/selected modifiers already within MDC List, which will be the first thing to use these mixins.

**Pull request merged**